IntegrationTests/conftest.py

```diff
@@ -0,0 +1,105 @@
+
+# Environment variables
+DEBUG = False
+CLASSPATH = ""
+EXECUTABLE = ""
+GENERATE_REPORT_LOCATION = ""
+TEST_EXCEPTIONS = ""
+GENERATE_REPORT = False
+
+
+# Log data
+def pytest_runtest_logreport(report):
+    """
+    Increment the counters for what action was performed
+    """
+    # Global required here to access counters
+    # Not ideal but without the counters wouldn't work
+    global total_tests, tests_passed, tests_failed, tests_skipped  # pylint: disable=global-statement
+    if report.when == "call":  # only count test function execution, not setup/teardown
+        total_tests += 1
+        if report.passed:
+            tests_passed += 1
+        elif report.failed:
+            tests_failed += 1
+        elif report.skipped:
+            tests_skipped += 1
+
+
+# Run after all tests completed, Generate a report of failing and passing tests
+def pytest_sessionfinish(exitstatus):
+    """
+    Generate report based on test run
+    """
+    if GENERATE_REPORT:
+        print("Generating report for the test run")
+        os.makedirs(GENERATE_REPORT_LOCATION, exist_ok=True)
+
+        # To make the report useful it will add the tests which have failed-
+        # -unexpectedly to known_failures
+        # It will also remove those that have passed from any of those lists
+
+        for test_path in tests_passed_unexpectedly:
+            test = str(test_path)
+            if test in known_failures:
+                known_failures.remove(test)
+            if test in unsupported:
+                unsupported.remove(test)
+            if test in failing_as_unspecified:
+                failing_as_unspecified.remove(test)
+
+        if len(tests_failed_unexpectedly) != 0:
+            for test in tests_failed_unexpectedly:
+                known_failures.append(str(test))
+
+        # Generate a report_message to save
+        report_data = {
+            "summary": {
+                "tests_total": total_tests,
+                "tests_passed": tests_passed,
+                "tests_failed": tests_failed,
+                "tests_skipped": tests_skipped,
+                "pytest_exitstatus": exitstatus,
+                "note": "Totals include expected failures",
+            },
+            "unexpected": {
+                "passed": [str(test) for test in tests_passed_unexpectedly],
+                "failed": [str(test) for test in tests_failed_unexpectedly],
+            },
+            "environment": {
+                "executable": EXECUTABLE,
+                "classpath": CLASSPATH,
+                "test_exceptions": TEST_EXCEPTIONS,
+                "debug": DEBUG,
+                "generate_report_location": GENERATE_REPORT_LOCATION,
+            },
+            "known_failures": known_failures,
+            "failing_as_unspecified": failing_as_unspecified,
+            "unsupported": unsupported,
+            "do_not_run": do_not_run,
+        }
+        print(f"Report location {GENERATE_REPORT_LOCATION}/report.txt")
```

Comment:
Inconsistent to what you actually do below. `GENERATE_REPORT` being a file name, it avoids this issue. It also avoids hardcoding any file names at all.